lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -889,12 +923,12 @@ def apply_api_config_if_needed(self) -> None:
         #     (3) based on the response, set the message status
         if state != BackupServiceState.SUCCESS:
             logger.error(f"Failed to setup backup service with state {state}")
-            if self.charm.unit.is_leader():
-                self.charm.status.set(BlockedStatus(BackupSetupFailed), app=True)
             self.charm.status.clear(BackupConfigureStart)
-            return
-        if self.charm.unit.is_leader():
-            self.charm.status.clear(BackupSetupFailed, app=True)
+            self.charm.status.set(BlockedStatus(BackupSetupFailed))
+            self.charm.status.set(BlockedStatus(BackupSetupFailed), app=True)
```

Comment:
Following conversation on: https://chat.canonical.com/canonical/pl/oqwsfdejufgepettcqzobnw7xh

It is neither guaranteed nor wanted by the entire team to support a: all units in a given status -> app status updated automatically